commands.go

```diff
@@ -1190,14 +1190,28 @@ func (c cmdable) LPop(ctx context.Context, key string) *StringCmd {
 	return cmd
 }
 
-func (c cmdable) LPos(ctx context.Context, key string, value string, rank int64, maxlen int64) *IntCmd {
-	cmd := NewIntCmd(ctx, "lpos", key, value, "rank", rank, "maxlen", maxlen)
+type LPosArgs struct {
+	Rank, MaxLen int64
+}
+
+func (c cmdable) LPos(ctx context.Context, key string, value string, args *LPosArgs) *IntCmd {
+	var rank, maxLen int64 = 1, 0
+	if args != nil {
+		rank = args.Rank
+		maxLen = args.MaxLen
+	}
```

Comment:
@vmihailenco What do you think of this? Setting sensible defaults.

Signature:
```go 
LPosCount(ctx context.Context, key string, value string, count int64, args *LPosArgs) 
```
Simplest use case looks like this:
```go
client.LPos(ctx, "list", "b", nil)
```
If the user needs more control he/she would need to pass in a LPosArgs struct

```go
client.LPos(ctx, "list", "b", &redis.LPosArgs{Rank: 2, MaxLen: 1})
```

Reply:
Perhaps it is convenient for your use case but it is counter-intuitive that 

```go
client.LPos(ctx, "list", "b", nil) // rank=1
client.LPos(ctx, "list", "b", &redis.LPosArgs{MaxLen: 1}) // rank=0
```

I believe it should be

```go
client.LPos(ctx, "list", "b", redis.LPosArgs{}) // lpos key value
client.LPos(ctx, "list", "b", redis.LPosArgs{MaxLen: 1}) // lpos key value maxlen 1
client.LPos(ctx, "list", "b", redis.LPosArgs{Rank: 1}) // lpos key value rank 1
client.LPos(ctx, "list", "b", redis.LPosArgs{Rank: 1, MaxLen: 1}) // lpos key value rank 1 maxlen 1
```

Reply:
Yeah... you're right, that does look odd. 

My reasoning was that if you send `&redis.LPosArgs{MaxLen: 1}` you should know that you're also sending `Rank:0` because of how Go "zero values" work and 0 is an invalid value for `Rank`. Sending `nil` would state more clearly that you don't want to set neither `Rank` nor `MaxLen`. In Redis not setting `Rank` is the same as having `Rank=1`, in other words, we could say that the default value for `Rank` in redis is `1`.

In your example, what do you think it's the best option to make `client.LPos(ctx, "list", "b", redis.LPosArgs{})` not fail. Check for `Rank == 0` and set it to `1` or check for `Rank == 0` and not set it?

Reply:
If `Rank = 0` is not useful/allowed then I expect that `redis.LPosArgs{}` does not set the rank. That applies even if we use `*redis.LPosArgs` and allow nil. E.g. `nil` and `&redis.LPosArgs{}` should both omit rank if `Rank = 0` does not make sense.

>Sending nil would state more clearly that you don't want to set neither Rank nor MaxLen

I am fine with allowing `nil` if you are more comfortable with it. But I believe the API should behave the same since AFAIK `Rank=0` and `MaxLen=0` does not make any sense

```go
client.LPos(ctx, "list", "b", nil) // lpos key value
client.LPos(ctx, "list", "b", &redis.LPosArgs{}) // lpos key value
client.LPos(ctx, "list", "b", &redis.LPosArgs{MaxLen: 1}) // lpos key value maxlen 1
client.LPos(ctx, "list", "b", &redis.LPosArgs{Rank: 1}) // lpos key value rank 1
client.LPos(ctx, "list", "b", &redis.LPosArgs{Rank: 1, MaxLen: 1}) // lpos key value rank 1 maxlen 1
```

I am leaning towards `redis.LPosArgs{}` since `nil` value does not have special meaning here. But it is probably not important enough.

>check for Rank == 0 and not set it?

This!

Reply:
Thanks! 

I'll try to sort it out later today.